pkg/ingester/client/client.go

```diff
@@ -64,14 +69,16 @@ func (c *closableHealthAndIngesterClient) Close() error {
 
 // Config is the configuration struct for the ingester client
 type Config struct {
-	GRPCClientConfig grpcclient.Config    `yaml:"grpc_client_config" doc:"description=Configures the gRPC client used to communicate with ingesters from distributors, queriers and rulers."`
-	CircuitBreaker   CircuitBreakerConfig `yaml:"circuit_breaker"`
+	GRPCClientConfig      grpcclient.Config    `yaml:"grpc_client_config" doc:"description=Configures the gRPC client used to communicate with ingesters from distributors, queriers and rulers."`
+	CircuitBreaker        CircuitBreakerConfig `yaml:"circuit_breaker"`
+	ReportGRPCStatusCodes bool                 `yaml:"report_grpc_status_codes"`
```

Comment:
I'd suggest to mark this as `category:"advanced"`.

Reply:
I have renamed the CLI flag to `-ingester.client.report-grpc-codes-in-instrumentation-label-enabled` and marked it as advanced.